mix.exs

```diff
@@ -14,9 +14,10 @@ defmodule OpenTelemetry.MixProject do
       # source_url: "https://github.com/USER/PROJECT",
       # homepage_url: "http://YOUR_PROJECT_HOMEPAGE",
       docs: [
-        main: "OpenTelemetry"
+        markdown_processor: ExDoc.Markdown.Cmark,
+        main: "OpenTelemetry",
         # logo: "path/to/logo.png",
-        # extras: ["README.md"]
+        extras: erlang_docs()
       ]
```

Comment:
There should be probably the task that would run `:edown_doclet` before running `mix docs`.

Reply:
Yea. Probably easier to do it in rebar3, have a post docs hook that runs mix docs. Then they can be published with rebar3.

I can't remember if there is a reason to publish to hex with mix over rebar3... I feel like there might have been, in which case we do need to have it as a "hook" in mix and not rbear3.

Reply:
No, it shouldn't be harder:

```elixir
aliases: [
  docs: ["cmd rebar3 edoc", "docs"]
]
```

Should do the job.

Reply:
Ah, aliases, nice.

And the alias is properly used even when something like `mix hex.publish` triggers a build?

Reply:
I have no guarantees, but I think it should do.

Reply:
Hm, I was going to say I'd refer publishing with rebar3 since it has `rebar3 hex cut`, and I don't think mix has similar? But that doesn't (yet) support bumping the version in a file like we are doing. So I guess we can go with mix.

Assuming the alias is not ignored when hex runs docs.

Reply:
No there is no such thing. However I have written script that can help with that:

```elixir
defmodule GitVersion do
  require Logger

  def version do
    case System.cmd(
           "git",
           ~w[describe --match=v* --always --dirty=+dirty],
           stderr_to_stdout: true
         ) do
      {"v" <> raw, 0} ->
        {:ok, version} = Version.parse(raw)

        version
        |> bump_version()
        |> to_string()

      {ver, 0} ->
        "0.0.0-#{String.trim(ver)}"
    end
  end

  defp bump_version(%Version{pre: []} = version), do: version

  defp bump_version(%Version{patch: p} = version),
    do: struct(version, patch: p + 1)
end
```

You can update it a little to provide such command. If I will find some time I can write it on my own.

Reply:
I checked, and if you have a `docs` alias it will be used by mix.

Reply:
Cool, thanks. I've updated this PR and included hex metadata.

Not going to worry about version bumping. The above code would still have to also create and push the git tag to work.

If I get the time to update `rebar3 hex cut` to support updating the VERSION file I might send a PR and suggest we switch to publishing the hex package with rebar3 instead. It should be much simpler than the `git` version of `cut` since it just reads and writes a file, so probably will come at some point :)

For now, just keeping all metadata in `mix.exs`, so at least for hex metadata they don't need to be synced between `mix.exs` and `rebar.config`.